internal/consensus/state.go

```diff
@@ -2090,6 +2090,13 @@ func (cs *State) defaultSetProposal(proposal *types.Proposal, recvTime time.Time
 	// TODO: We can check if Proposal is for a different block as this is a sign of misbehavior!
 	if cs.ProposalBlockParts == nil {
 		cs.ProposalBlockParts = types.NewPartSetFromHeader(proposal.BlockID.PartSetHeader)
+
+		// If we are the proposer, lock the PartSet until we load all
+		// the block parts, that should come just after this Proposal.
+		// Addresses: https://github.com/cometbft/cometbft/issues/1742
+		if bytes.Equal(pubKey.Bytes(), cs.privValidatorPubKey.Bytes()) {
```

Comment:
I know you didn't introduce it, but, could you rename variable `pubKey` to `proposerPubKey` to make the code less confusing?